### PR TITLE
Add common API calls

### PR DESCRIPTION
**Pull request by juliamagan**
## Description

In this PR we have added all the endpoints that were previously used in the system tests, and some new ones that may be useful.

`WazuhHandler` will have a `WazuhAPI` object as an attribute, and will be able to call requests when necessary.

### CODE REVIEW

File: src/wazuh_qa_framework/wazuh_components/api/wazuh_api.py [hunk 1]
```diff
@@ -113,12 +113,468 @@ def list_agents(self, response):
...
+    @WazuhAPIRequest(method='GET', endpoint='/agents')
+    def get_agent_id(self, response, host_ip):
+        """Get agent ID.
+
+        Returns:
+            str: Agent ID.
+        """
+        for agent in response.data['affected_items']:
+            if host_ip in agent['ip']:
+                return agent['id']
```
Comment (Rebits) [inline src/wazuh_qa_framework/wazuh_components/api/wazuh_api.py:125]: Should we raise an exception in case the agent is not registered?

**Review by Rebits — CHANGES_REQUESTED** [review]
Great job! Some changes are required

File: src/wazuh_qa_framework/wazuh_components/api/wazuh_api.py (hunk 1 above)
Comment (juliamagan) [reply src/wazuh_qa_framework/wazuh_components/api/wazuh_api.py]: Done in a0c5f44

File: src/wazuh_qa_framework/system/wazuh_handler.py [hunk 2]
```diff
@@ -501,26 +504,30 @@ def get_agents_info(self):
         Returns:
             dict: Agent information
         """
-        pass
+        return self.wazuh_api.list_agents()
 
-    def get_agents_id(self, agents_list=None):
-        """Get agents id
+    def get_agents_id(self, host):
```
Comment (Rebits) [inline src/wazuh_qa_framework/system/wazuh_handler.py:509]:
```suggestion
    def get_agent_id(self, host):
```

File: src/wazuh_qa_framework/system/wazuh_handler.py [hunk 3]
```diff
@@ -501,26 +504,30 @@ def get_agents_info(self):
...
 
         Returns:
-            List: Agents id list
+            str: Agent ID
         """
-        pass
+        return self.wazuh_api.get_agent_id(host_ip=self.get_host_ansible_ip(host))
 
-    def restart_agent(self, host):
+    def restart_agent(self, host, method='service'):
```
Comment (Rebits) [inline src/wazuh_qa_framework/system/wazuh_handler.py:517]: method param should be included in `restarts_agents` too

**Review by Rebits — CHANGES_REQUESTED** [review]
Great job! Some changes are required

File: src/wazuh_qa_framework/system/wazuh_handler.py (hunk 3 above)
Comment (juliamagan) [reply src/wazuh_qa_framework/system/wazuh_handler.py:517]: Done in ae31c81

**Pull request merged**